### PR TITLE
minor: Add basic editorconfig and pre-commit hooks to enforce style for whitespaces

### PR DESCRIPTION
**Pull request by XuehaiPan**
## Motivation

Add configs and tools to ensure code style consistency.

## Modifications

Add `.editorconfig` and some `pre-commit` hooks.

1. Ensure text files are ended with an empty line.
2. No tailing whitespaces in text files.

## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
@XuehaiPan We have already used `isort` and `black` for the format

**Comment by zhyncs** [comment]
For other config file(yml, json, etc) and doc(rst, md, etc), use editor config is ok for me

**Comment by zhyncs** [comment]
@XuehaiPan Could you run the pre-commit locally to get a formatted version

**Comment by XuehaiPan** [comment]
> @XuehaiPan We have already used `isort` and `black` for the format

@zhyncs `black` is not enabled due to incorrect `types` in pre-commit config: `black`: not files to check

```console
$ git checkout main
$ pre-commit run --all-files
isort....................................................................Passed
black................................................(no files to check)Skipped
don't commit to branch...................................................Failed
- hook id: no-commit-to-branch
- exit code: 1
```

I can enable it in a follow-up PR. Or we can migrate `isort + black` to `ruff format`.

**Comment by zhyncs** [comment]
@XuehaiPan We can only use the editor config for configuration files and documentation, not for Python code.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+default_stages: [pre-commit, pre-push, manual]
+
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
```
Comment (XuehaiPan) [inline .pre-commit-config.yaml:13]: > not for Python code.

The editorconfig file is compatible with `black` style. As commented in https://github.com/sgl-project/sglang/pull/1926#issuecomment-2459066525, `black` is not enabled in our CI.

The `.editorconfig` file only defines a convention for the editor (e.g. vscode). It does not change the code files. The changes for Python code are made by `pre-commit` hook `trailing-whitespace` and `end-of-file-fixer`.

**Review by zhyncs — APPROVED** [review]

**Pull request merged**